sample/proguard-project.pro

```diff
@@ -0,0 +1,163 @@
+# This file contains project specific proguard rules
+
+-android
+
+# Print out the full proguard config used for every build
+-printconfiguration build/outputs/fullProguardConfig.pro
+
+# These classes are used via kotlin reflection and the keep might not be required anymore once Progurad supports
+# Kotlin reflection directlty.
+-keep class kotlin.reflect.jvm.internal.impl.builtins.BuiltInsLoaderImpl
+-keep class kotlin.reflect.jvm.internal.impl.load.java.FieldOverridabilityCondition
+-keep class kotlin.reflect.jvm.internal.impl.load.java.ErasedOverridabilityCondition
+-keep class kotlin.reflect.jvm.internal.impl.load.java.JavaIncompatibilityRulesOverridabilityCondition
+
+# If Companion objects are instantiated via Kotlin reflection and they extend/implement a class that Proguard
+# would have removed or inlined we run into trouble as the inheritance is still in the Metadata annoation 
+# read by Kotlin reflection.
+# FIXME Remove if Kotlin reflection is supported by Pro/Dexguard
+-if class **$Companion extends **
+-keep class <2>
+-if class **$Companion implements **
+-keep class <2>
+
+# https://medium.com/@AthorNZ/kotlin-metadata-jackson-and-proguard-f64f51e5ed32
+-keep class kotlin.Metadata { *; }
+
+# https://stackoverflow.com/questions/33547643/how-to-use-kotlin-with-proguard
+-dontwarn kotlin.**
+
+# https://app.bugsnag.com/airbnb/android-1/errors/59f6c07487a5c5001a49d882?filters[event.since][0]=all&filters[event.severity][0][value]=error&filters[event.severity][0][type]=eq
+# https://github.com/square/okhttp/issues/3582
+-keep class okhttp3.internal.publicsuffix.PublicSuffixDatabase
+
+# The code is safeguarded against API28 use so we can just ignore the warning bout Handler.createAsync(looper) not existing.
+# Can be removed when compiled against API >=28
+-dontwarn com.airbnb.epoxy.EpoxyAsyncUtil
+
+# RxJava
+-keepclassmembers class rx.internal.util.unsafe.*ArrayQueue*Field* {
```

Comment:
Is this really needed? I thought RxJava already worked great with ProGuard.

Reply:
We found that we needed that in our app, it might not be needed here (so is the full config for retrofit, moshi and okhttp3) it is more here for completeness, as this app uses RxJava, it should have the config that is needed for full support.

Reply:
Oh, I see, you just copied the Proguard from Airbnb?

Reply:
I did not, our file is way more complicated. :-) 

I just copied the sections for the libs that overlap and are not taken care of the libs themselves.